internal/common/slices/slices.go

```diff
@@ -1,9 +1,69 @@
 package slices
 
+import (
+	"fmt"
+	"math"
+
+	goslices "golang.org/x/exp/slices"
+)
+
 func Map[T any, U any](list []T, fn func(val T) U) []U {
 	out := make([]U, len(list))
 	for i, val := range list {
 		out[i] = fn(val)
 	}
 	return out
 }
+
+// PartitionToLen partitions the elements of s into non-overlapping slices,
+// such that each such slice contains at most maxLen elements.
+func PartitionToMaxLen[S ~[]E, E any](s S, maxLen int) []S {
+	n := int(math.Ceil(float64(len(s)) / float64(maxLen)))
+	if n == 0 {
+		n = 1
+	}
+	return Partition(s, n)
+}
+
+// Partition partitions the elements of s into n non-overlapping slices,
```

Comment:
Do we need the behaviour that some slices are len/n and some len/n+1?  The more common behaviour is to make a slices length 1/n and the last slice have between 1 and 1/n elements.  That has the benefit that the implementation is simpler.

Reply:
When used for load-balancing this would be slightly better. But it may be an unnecessary optimisation. We don't need these functions at all anymore and could remove them entirely.